### ISSUE TITLE
AOT compiler not found when building for iOS

### ISSUE THREAD

**Issue by Djakar** [dotnet/maui#8106]
### Description

When I try to run on iOS simulator I got an error (sorry, it is french):
Erreur		Le compilateur AOA'  'n’existe pas.	Vega.App.Infi	C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.414\targets\Xamarin.Shared.Sdk.targets	973	

Note that 'AOA' means AOT

Translated in english, it say something like: AOT compiler not found

### Steps to Reproduce

1. Create MAUI app on Windows 11.
2. Choose any iOS Simulator
3. Run the app

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15.5

### Did you find any workaround?

No workaround found

### Relevant log output

```shell
L'opération de génération a démarré...
1>------ Début de la génération : Projet : MauiApp1, Configuration : Debug Any CPU ------
1>Vous utilisez une préversion de .NET. Consultez https://aka.ms/dotnet-core-preview
1>Executing SayHello Task to establish a connection to a Remote Server. 
1>			Properties: 
1>				SessionId=01f6b418d615d12802b9512876aa800378e22130f8cb533eeff30787b0637f57, 
1>				Addresss=192.168.0.172, 
1>				SshPort=22, 
1>				TcpPort=53144, 
1>				User=csanchez, 
1>				AppName=MauiApp1,
1>				VisualStudioProcessId=26252,
1>				ContinueOnDisconnected=False
1>Executing SayHello Task to establish a connection to a Remote Server. 
1>			Properties: 
1>				SessionId=01f6b418d615d12802b9512876aa800378e22130f8cb533eeff30787b0637f57, 
1>				Addresss=192.168.0.172, 
1>				SshPort=22, 
1>				TcpPort=53144, 
1>				User=csanchez, 
1>				AppName=MauiApp1,
1>				VisualStudioProcessId=26252,
1>				ContinueOnDisconnected=False
1>Identité de signature détectée :
1>  Bundle Id: com.companyname.mauiapp1
1>  App Id: com.companyname.mauiapp1
1>Les analyseurs ont été ignorés pour accélérer la génération. Vous pouvez exécuter la commande « Build » ou « Rebuild » pour exécuter des analyseurs.
1>MauiApp1 -> E:\MauiApp1\MauiApp1\bin\Debug\net6.0-ios\iossimulator-x64\MauiApp1.dll
1>L'exécution de l'outil /Users/csanchez/Library/Caches/Xamarin/XMA/SDKs/dotnet/dotnet s'est achevé (code de sortie = 1).
1>Microsoft (R) Build Engine version 17.3.0-preview-22277-01+f1dae6ab6 for .NET
1>Copyright (C) Microsoft Corporation. All rights reserved.
1>
1>  Determining projects to restore...
1>  Retrying 'FindPackagesByIdAsyncCore' for source 'https://registry.vega-logiciel.fr/repository/master-group-nuget/FindPackagesById()?id='System.Runtime.InteropServices.NFloat.Internal'&semVerLevel=2.0.0'.
1>  Response status code does not indicate success: 404 (Not Found).
1>  Retrying 'FindPackagesByIdAsyncCore' for source 'https://registry.vega-logiciel.fr/repository/master-group-nuget/FindPackagesById()?id='System.Runtime.InteropServices.NFloat.Internal'&semVerLevel=2.0.0'.
1>  Response status code does not indicate success: 404 (Not Found).
1>/var/folders/8m/l3vg6j6x2c7c5h920_zjk6fw0000gn/T/tmp5cf23a3d.tmp.csproj : error NU1301: Failed to retrieve information about 'System.Runtime.InteropServices.NFloat.Internal' from remote source 'https://registry.vega-logiciel.fr/repository/master-group-nuget/FindPackagesById()?id='System.Runtime.InteropServices.NFloat.Internal'&semVerLevel=2.0.0'.
1>  Failed to restore /var/folders/8m/l3vg6j6x2c7c5h920_zjk6fw0000gn/T/tmp5cf23a3d.tmp.csproj (in 2.93 sec).
1>
1>Build FAILED.
1>
1>/var/folders/8m/l3vg6j6x2c7c5h920_zjk6fw0000gn/T/tmp5cf23a3d.tmp.csproj : error NU1301: Failed to retrieve information about 'System.Runtime.InteropServices.NFloat.Internal' from remote source 'https://registry.vega-logiciel.fr/repository/master-group-nuget/FindPackagesById()?id='System.Runtime.InteropServices.NFloat.Internal'&semVerLevel=2.0.0'.
1>    0 Warning(s)
1>    1 Error(s)
1>
1>Time Elapsed 00:00:03.91
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.414\targets\Xamarin.Shared.Sdk.targets(973,3): error : /Users/csanchez/Library/Caches/Xamarin/XMA/SDKs/dotnet/dotnet s'est arrêté. Code de sortie : 1
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.414\targets\Xamarin.Shared.Sdk.targets(973,3): error :
1>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\15.4.414\targets\Xamarin.Shared.Sdk.targets(973,3): error : Le compilateur AOA'  'n’existe pas.
1>Génération du projet "MauiApp1.csproj" terminée -- ÉCHEC.
========== Build : 0 réussite(s), 1 échec(s), 0 à jour, 0 ignorée(s) ==========
========== Déploiement : 0 réussis, 0 échoués, 0 ignorés ==========
```

**Comment by jfversluis** [comment]
I think the actual error is some NuGet package that it can't resolve. See the NU1301 error in there. Please disable any custom NuGet feeds that you might have setup and try again.

**Comment by Djakar** [comment]
Hello and thank you for your response.

I did see the nuget error but it was totally incomprehensible because I removed all nuget sources on my windows. I checked the nuget.config file for this and left only nuget.org.

Despite everything, this error was persistent.
I ended up checking the nuget sources on the mac and indeed I saw an old nuget source that I no longer had access to.

So I fixed the problem by restoring the nuget.org source on the mac.

What I didn't understand was that all the packages consumed by my MAUI app were being resolved on my windows PC but one package is being picked up by MS Build targets directly from the mac.
